index.js

```diff
@@ -3,11 +3,26 @@ module.exports = function (str, opts) {
 	str = str.toString();
 	opts = opts || {};
 
-	var preserve = !opts.all;
+	var comment = '';
 	var currentChar = '';
 	var insideString = false;
+	var preserveFilter;
+	var preserveImportant = (opts.preserve === false || opts.all === true) ? false : true;
 	var ret = '';
 
+	//use preserveFilter as an interface for opts.preserve function and RegExp values
+	if (opts !== undefined && opts.preserve !== undefined) {
+		if ((typeof opts.preserve).toLowerCase() === 'function') {
+			preserveImportant = false;
+			preserveFilter = opts.preserve;
+		} else if ((opts.preserve.constructor.name).toLowerCase() === 'regexp') {
```

Comment:
@Qix- , thanks for the recommendation! I wasn't aware of the performance consideration.

@sindresorhus, please see below about assuming `RegExp`.